include/PriorityQueue.h

```diff
@@ -4,16 +4,59 @@
 #include <list>
 #include <exception>
 
+/**
+ * @brief The MyComparator struct
+ * @details This struct is used to compare two objects of the same type.
+ * The operator() is used to compare two objects of the same type.
+ */
+template <class T>
+struct MyComparator {
+    T operator()(T a, T b) const {
+        return a - b;
+    }
+};
 
-// any templates?
+/**
+ * @brief The PriorityQueue class
+ * @details This class is a priority queue. It is a list of objects of the same type.
+ * The objects are sorted by the MyComparator struct.
+ */
+template <class T>
 class PriorityQueue {
 
 public:
-
-    // You need to complete the implement :
     void push(const T& t);
     T poll();
 
 private:
-// add relevant data members
+    std::list<T> m_pq;
 };
+
+/**
+ * @brief push
+ * @details This method is used to insert an object in the priority queue.
+ * @param t The object to insert.
+ */
+template<class T>
+inline void PriorityQueue<T>::push(const T& t)
+{
+    auto it = m_pq.begin();

```

Comment:
When possible it's recommended to call cbegin() and cend(). You can never be too safe